### DISCUSSION TITLE
[QUESTION] AttributeError: '_io.BufferedReader' object has no attribute 'parse_mode' on Bot.edit_message_media

### DISCUSSION THREAD

**Discussion by GivenFLY** [python-telegram-bot/python-telegram-bot#2405] [Q&A] [answered]
Code example:
```py
from time import sleep
from telegram import Bot

token = "xxxxxxx:yyyyyyyyyyyyyyy"
tgid = 0
path1 = "1.png"
path2 = "2.png"

message = Bot(token).send_photo(tgid, open(path1, "rb"))
sleep(2)
Bot(token).edit_message_media(tgid, message.message_id, media=open(path2, "rb"))
````

Bot should send first picture, and after 2 secconds change it to seccond
Actually bot sends first pic but while it changes to seccond an exception appears:
```
Traceback (most recent call last):
  File ".\issue.py", line 11, in <module>
    Bot(token).edit_message_media(tgid, message.message_id, media=open(path2, "rb"))
  File "<decorator-gen-29>", line 2, in edit_message_media
  File "C:\Users\GivenFLY\AppData\Local\Programs\Python\Python38-32\lib\site-packages\telegram\bot.py", line 67, in decorator
    result = func(*args, **kwargs)
  File "C:\Users\GivenFLY\AppData\Local\Programs\Python\Python38-32\lib\site-packages\telegram\bot.py", line 2020, in edit_message_media
    return self._message(url, data, timeout=timeout, reply_markup=reply_markup, **kwargs)
  File "C:\Users\GivenFLY\AppData\Local\Programs\Python\Python38-32\lib\site-packages\telegram\bot.py", line 169, in _message
    if data.get('media') and (data['media'].parse_mode == DEFAULT_NONE):
AttributeError: '_io.BufferedReader' object has no attribute 'parse_mode'
```

### Configuration
**Operating System: Windows 10**
```
python-telegram-bot 12.8
certifi 2020.06.20
Python 3.8.5 (tags/v3.8.5:580fbb0, Jul 20 2020, 15:43:08) [MSC v.1926 32 bit (Intel)]
```

**Comment by Bibo-Joshi** [comment]
Hi. As you can see in the docs of [`edit_message_media`](https://python-telegram-bot.readthedocs.io/en/stable/telegram.bot.html#telegram.Bot.edit_message_media), the `media` argument must be of type [`InputMedia`](https://python-telegram-bot.readthedocs.io/en/stable/telegram.inputmedia.html#telegram.InputMedia),  in your case probaby [`InputMediaPhoto`](https://python-telegram-bot.readthedocs.io/en/stable/telegram.inputmediaphoto.html#telegram.InputMediaPhoto).
BTW, instead of instatiating a new `Bot` instance for every method call, you can just do
```python
bot = Bot(token)
message = bot.send_photo(…)
bot.edit_message_media
```
or better yet, just call [`message.edit_media`](https://python-telegram-bot.readthedocs.io/en/stable/telegram.message.html#telegram.Message.edit_media)

**Comment by GivenFLY** [comment]
But doesn't InputMediaPhoto inherit from InputMedia? 
In my case I can't simply use message.edit_media. I need to change a picture using message_id

Even if I use message.edit_media it shows me same exceprion

```py
bot = Bot(token)
message = bot.send_photo(tgid, open(path1, "rb"))
sleep(2)
message.edit_media(media=open(path2, "rb"))
```
```
Traceback:
......
AttributeError: '_io.BufferedReader' object has no attribute 'parse_mode'
```

**Comment by Bibo-Joshi** [comment, marked as answer]
> But doesn't InputMediaPhoto inherit from InputMedia?

yes. that's the point ;)

> In my case I can't simply use message.edit_media. I need to change a picture using message_id

As per the docs I pointed you to, `message.edit_media(**kwargs)` is equivalent to `bot.edit_message_media(chat_id=message.chat_id, message_id=message.message_id, **kwargs)`

> Even if I use message.edit_media it shows me same exceprion

because you still pass an open file instead of an `InputMediaPhoto` instance ;)

**Reply by GivenFLY** [reply]
But how to pass an InputMedia instance?

**Reply by Bibo-Joshi** [reply]
`media=InputMediaPhoto(…)`

**Reply by GivenFLY** [reply]
Thank you. It helps me :)
